### ISSUE TITLE
polyclang fails when PATH is not explicitly set

### ISSUE THREAD

**Issue by bradlarsen** [trailofbits/polytracker#23]
If you try to build something with `polyclang` without explicitly modifying your `PATH`, you might get strange linking errors. For example:
```
$ echo $PATH
/home/blarsen/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin

$ /home/blarsen/polytracker/build/bin/polytracker/polyclang++ -Wall -O2 /home/blarsen/polytracker/polytracker/test/test1.c -o test1
clang++ -Wall -O2 /home/blarsen/polytracker/polytracker/test/test1.c -o test1 -Xclang -load -Xclang /home/blarsen/polytracker/build/bin/polytracker/pass/libDataFlowSanitizerPass.so -mllvm -polytrack-dfsan-abilist=/home/blarsen/polytracker/build/bin/polytracker/abi_lists/polytrack_abilist.txt -mllvm -polytrack-dfsan-abilist=/home/blarsen/polytracker/build/bin/polytracker/abi_lists/dfsan_abilist.txt -Wl,--start-group -Wl,--whole-archive /home/blarsen/polytracker/build/bin/polytracker/lib/libdfsan_rt-x86_64.a -Wl,--no-whole-archive -Wl,--dynamic-list=/home/blarsen/polytracker/build/bin/polytracker/lib/libdfsan_rt-x86_64.a.syms /home/blarsen/polytracker/build/bin/polytracker/lib/libTaintSources.a -Wl,--no-as-needed -Wl,--gc-sections -ldl -lpthread -lm -Wl,--end-group -pie -fpic -Qunused-arguments
clang: warning: treating 'c' input as 'c++' when in C++ mode, this behavior is deprecated [-Wdeprecated]
/home/blarsen/polytracker/build/bin/polytracker/lib/libdfsan_rt-x86_64.a(sanitizer_posix_libcdep.cc.o): In function `__sanitizer::GetNamedMappingFd(char const*, unsigned long)':
/home/blarsen/polytracker/polytracker/dfsan/dfsan_rt/sanitizer_common/sanitizer_posix_libcdep.cc:321: undefined reference to `shm_open'
/home/blarsen/polytracker/polytracker/dfsan/dfsan_rt/sanitizer_common/sanitizer_posix_libcdep.cc:325: undefined reference to `shm_unlink'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

However, in this example, if I append `/home/blarsen/polytracker/build/bin/polytracker` to my `PATH`, `polyclang` succeeds.

It would be better if your `PATH` didn't need to be modified to be able to use `polyclang` or `polyclang++`.

**Comment by bradlarsen** [comment]
The linker errors about missing symbols is in fact probably due to a bug in polyclang where `-lgcc_s` isn't passed when it should be.

**Comment by bradlarsen** [comment]
Seemingly fixed by the merge of PR #28.
